src/Integration.Vsix/SonarLintTagger/Scheduler.cs

```diff
@@ -38,29 +38,26 @@ public Scheduler()
             jobs = new Dictionary<string, WeakReference<CancellationTokenSource>>(StringComparer.OrdinalIgnoreCase);
         }
 
-        public void Schedule(string jobId, Action<CancellationToken> action, int? timeoutInMilliseconds)
+        public void Schedule(string jobId, Action<CancellationToken> action, int timeoutInMilliseconds)
         {
-            var newTokenSource = IssueToken(jobId);
+            var newCancellationToken = IssueToken(jobId, timeoutInMilliseconds);
 
-            if (timeoutInMilliseconds.HasValue)
-            {
-                newTokenSource.CancelAfter(timeoutInMilliseconds.Value);
-            }
-
-            action(newTokenSource.Token);
+            action(newCancellationToken);
             // The job might be running asynchronously so we don't know when to dispose the CancellationTokenSources, and have to rely on weak-refs and garbage collection to do it for us
         }
 
-        private CancellationTokenSource IssueToken(string jobId)
+        private CancellationToken IssueToken(string jobId, int timeoutInMilliseconds)
         {
             lock (jobs)
             {
                 CancelPreviousJob(jobId);
 
                 var newTokenSource = new CancellationTokenSource();
+                newTokenSource.CancelAfter(timeoutInMilliseconds);
```

Comment:
I can see how the race condition existed before since the `newTokenSource` was being accessed outside the lock.
However, I don't understand why the timeout needs to be mandatory to fix the bug.

Reply:
Yep, existed before but was much less likely to happen.
We will always pass a timeout now, so we might as well make it non-nullable.

Reply:
Is there any point in calling `CancelAfter` with an infinite timeout?

Reply:
Are you suggesting to do `if timeoutInMilliseconds !=Timeout.Infinite`? 
I think it can be a large timeout anyway.

Reply:
I was thinking about skipping the call to `CancelAfter`. However, since we're setting a non-infinite timeout in the product code, the only time this could be infinite would be if the user set it via the environment variable. And the validation logic for the environment variable ignores values < 1, so the user couldn't set it to infinite either.
So there's no point in changing anything here.